easybuild/framework/easyconfig/tweak.py

```diff
@@ -833,6 +950,71 @@ def map_easyconfig_to_target_tc_hierarchy(ec_spec, toolchain_mapping, targetdir=
                 # set module names
                 orig_dep['short_mod_name'] = ActiveMNS().det_short_module_name(dep)
                 orig_dep['full_mod_name'] = ActiveMNS().det_full_module_name(dep)
+            elif update_dep_versions:
+                # Search for available updates for this dependency:
+                # First get all candidate paths for this (include search through subtoolchains)
+                toolchain_hierarchy = get_toolchain_hierarchy(orig_dep['toolchain'])
+                potential_version_matches = []
+
+                # Figure out what precedes the version
+                if 'versionprefix' in dep:
+                    versionprefix = dep['versionprefix']
+                else:
+                    versionprefix = ''
+                versionprefix = ''.join([x for x in [dep['name'] + '-', versionprefix] if x])
+                # Figure out the main versionsuffix (altered depending on toolchain in the loop below)
+                if 'versionsuffix' in dep:
+                    if dep['versionsuffix'] in versonsuffix_mapping:
+                        versionsuffix = versonsuffix_mapping[dep['versionsuffix']]
+                    else:
+                        versionsuffix = dep['versionsuffix']
+                else:
+                    versionsuffix = ''
```

Comment:
This whole block can be collapsed to a single line as well:

```python
versionsuffix = versonsuffix_mapping.get(dep.get('versionsuffix', None), '')
```

If there is no `versionsuffix`, `dep.get('versionsuffix', None)` will return `None`, which won't be included in `versonsuffix_mapping`, so you'll get the fallback value `''`.

You can even just use `dep.get('versionsuffix')` rather than `dep.get('versionsuffix', None)`, since `None` is the default for `.get()` on a dict value:

```python
>>> print {}.get('foo')
None